python/whylogs/core/schema.py

```diff
@@ -40,30 +39,41 @@ class DatasetSchema:
     >>> import pandas as pd
     >>> import numpy as np
     >>> from whylogs.core import DatasetSchema, DatasetProfile
+    >>> from whylogs.core.resolvers import Resolver, StandardResolver
     >>>
-    >>> class MySchema(DatasetSchema):
-    ...    types = {
+    >>> class MyResolver(StandardResolver):
+    ...    pass
+    >>>
+    >>> schema = DatasetSchema(
+    ...    types={
     ...        "col1": str,
     ...        "col2": np.int32,
-    ...        "col3": pd.Categorical(categories=('foo', 'bar'), ordered=True)
-    ...    }
-    ...    cache_size = 12
-    >>> schema = MySchema()
-    >>> prof = DatasetProfile(MySchema())
+    ...        "col3": pd.CategoricalDtype(categories=('foo', 'bar'), ordered=True)
+    ...    },
+    ...    resolvers=MyResolver()
```

Comment:
Now the user will need to get their own resolver correct? How did this fix the problem you were encountering before?

Reply:
The 2 problems were
1. Users had to use `field(default_factory=...)` to override the defaults when subclassing `DatasetSchema`
2. `DatasetSchema::copy()` always returned `DatasetSchema` rather than the dynamic type of the copied schema

To fix 1, we're essentially saying "don't subclass `DatasetSchema`, just pass your `types` and `resolvers` into the constructor." If one did wish to subclass `DatasetSchema`, one would have to override `__init__()` to change the defaults, which is probably worse than using `field()`. But my impression is that users mostly only want to subclass `Resolver`, and passing that into the `DatasetSchema` constructor is probably better than subclassing `DatasetSchema` just to use a different `Resolver`.

Problem 2 is fixed by fixing it :) just in case users still subclass `DatasetSchema`.

Reply:
That makes sense :) I agree.